src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -771,20 +770,20 @@ namespace winrt::TerminalApp::implementation
     TerminalConnection::ITerminalConnection TerminalPage::_CreateConnectionFromSettings(GUID profileGuid,
                                                                                         TerminalApp::TerminalSettings settings)
     {
-        const auto* const profile = _settings->FindProfile(profileGuid);
+        const auto profile = _settings->FindProfile(profileGuid);
 
         TerminalConnection::ITerminalConnection connection{ nullptr };
 
         GUID connectionType{ 0 };
         GUID sessionGuid{ 0 };
 
-        if (profile->HasConnectionType())
+        if (profile.ConnectionType() != nullptr)
         {
-            connectionType = profile->GetConnectionType();
+            connectionType = profile.ConnectionType().Value();
         }
 
-        if (profile->HasConnectionType() &&
-            profile->GetConnectionType() == AzureConnectionType &&
+        if (profile.ConnectionType() != nullptr &&
```

Comment:
Should we just cache the result of `profile.ConnectionType() != nullptr` in a local bool in this function and re-use that one around here? (same with using `connectionType` when appropriate)